src/cls/tabular/fbwriter_skyroot.cc

```diff
@@ -0,0 +1,784 @@
+    auto rows = Tables::CreateRows(
+      builder,
+      0,
+      0,
+      table_name,
+      schema_fb,
+      nrows,
+      ncols,
+      layout,
+      rids_vect_fb,
+      row_records_fb ) ;
+
+    Tables::RootBuilder root_builder( builder ) ;
+    //root_builder.add_schema( a ) ;
+
+    // save the Rows flatbuffer to the root flatbuffer
+    root_builder.add_relationData_type( Tables::Relation_Rows ) ;
+    root_builder.add_relationData( rows.Union() ) ;
+
+    auto res = root_builder.Finish() ;
+    builder.Finish( res ) ;
+
+    const char* fb = reinterpret_cast<char*>( builder.GetBufferPointer() ) ;
+    int bufsz      = builder.GetSize() ;
+    librados::bufferlist bl;
+    bl.append( fb, bufsz ) ;
+    librados::bufferlist wrapper_bl ;
+    ::encode( bl, wrapper_bl ) ;
+
+    if( inputs.writeto == "ceph" ) {
+      writeToCeph( wrapper_bl, bufsz, inputs.targetoid, inputs.targetpool ) ;
+    }
+    else if( inputs.writeto == "disk" ) {
+      writeToDisk( wrapper_bl, bufsz, inputs.targetoid ) ;
+    } 
+    else {
+      std::cout << ">>> unrecognized writeto '" << inputs.writeto << "'" << std::endl ;
+      exit( 1 ) ;
+    }
+  }
+
+  // -----------------------------------------------
+  // | FBMeta flatbuffer     | Col flatbuffer     |
+  // -----------------------------------------------
+  else if( inputs.write_type == "col" ) {
+
+    // --------------------------------------------- //
+    // --------------------------------------------- //
+    uint64_t nrows = inputs.nrows ;
+    uint64_t ncols = inputs.ncols ;
+
+    // from commandline
+    // place these before record_builder declare
+    auto table_name     = builder.CreateString( inputs.table_name ) ;
+    auto layout         = builder.CreateString( inputs.write_type ) ;
+
+    // parse schema csv strings
+    std::vector< std::string > schema_attnames = parse_csv_str( inputs.schema_attnames ) ;
+    std::vector< flatbuffers::Offset< flatbuffers::String > > schema ;
+    for( int i = 0; i < schema_attnames.size(); i++ ) {
+      schema.push_back( builder.CreateString( schema_attnames[i] ) ) ;
+    }
+    auto schema_fb = builder.CreateVector( schema ) ;
+
+    std::vector< std::string > schema_datatypes = parse_csv_str( inputs.schema_datatypes ) ;
+    std::vector< uint8_t > record_data_type_vect ;
+    for( int i = 0; i < schema_datatypes.size(); i++ ) {
+      std::string this_dt = schema_datatypes[i] ;
+      if( this_dt == "int" )
+        record_data_type_vect.push_back( Tables::Data_IntData ) ;
+      else if( this_dt == "float" )
+        record_data_type_vect.push_back( Tables::Data_FloatData ) ;
+      else if( this_dt == "string" )
+        record_data_type_vect.push_back( Tables::Data_StringData ) ;
+      else {
+        std::cout << ">> unrecognized schema_datatype '" << this_dt << "'" << std::endl ;
+        exit( 1 ) ;
+      }
+    }
+    auto record_data_types = builder.CreateVector( record_data_type_vect ) ;
+
+    // establish rids_vect
```

Comment:
also a comment here?  i'm not sure why 3 rids first then loop nrows rids?  thanks.